libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -38,6 +38,12 @@ VoteManager::VoteManager(const addr_t& node_addr, const PbftConfig& pbft_config,
         continue;
       }
 
+      if (vote->getType() == PbftVoteTypes::cert_vote) {
+        reward_votes_block_hash_ = vote->getBlockHash();
+        reward_votes_period_ = vote->getPeriod();
+        reward_votes_round_ = vote->getRound();
+      }
```

Comment:
this will happen 2t+1 times...

Reply:
yes, basically all of the 2t+1 cert votes should have this value the same and this is done on startup. I will change it with some flag to do only once.